### PR TITLE
Changed modules & visitor mods to public, set filemap::write_file public

### PR DESCRIPTION
**Pull request by DanielJCampbell**
Changes to improve rustfmt usability as a library - FmtVisitor made public so users can format without having to plug into the main format methods in lib, write file is public to give more control over output filenames/locations, modules is public to allow access to list_files (used to format a file with the visitor).

### CODE REVIEW

File: src/filemap.rs [hunk 1]
```diff
@@ -80,7 +80,7 @@ pub fn write_system_newlines<T>(writer: T,
     }
 }
 
-fn write_file<T>(text: &StringBuffer,
+pub fn write_file<T>(text: &StringBuffer,
                  filename: &str,
```
Comment (nrc) [inline src/filemap.rs]: please adjust the indentation on following lines (surprised this doesn't fail tests)

**Comment by nrc** [comment]
> modules is public to allow access to list_files (used to format a file with the visitor)

might it be better to pull this function up rather than making the whole module public?

**Comment by DanielJCampbell** [comment]
@nrc, Currently the only content of that module are that method and two (non-public) helper methods, so the net effect is the same. Just tell me which you'd prefer.

**Comment by nrc** [comment]
Ok, then how it is in the PR seems fine

**Pull request merged**